gdx/src/com/badlogic/gdx/graphics/g2d/BitmapFont.java

```diff
@@ -824,6 +824,10 @@ public int getWrapIndex (Array<Glyph> glyphs, int start) {
 				char ch = (char)glyphs.get(i).id;
 				if (isWhitespace(ch) || isBreakChar(ch)) return i + 1;
 			}
+			for (i=0; i<start; i++) { // Move forward excluding whitespaces
+				char ch = (char)glyphs.get(i).id;
+				if (!isWhitespace(ch)) return i;
+			}
```

Comment:
This looks fishy. The code in this method:
1) Looks backward from `start - 1` to the first non-whitespace character. I don't remember exactly why.
2) From there, it continues looking backward to the first whitespace or break character. If found, it returns the index of the character after the whitespace or break character.
3) The new code starts looking from 0 and returns the first non-whitespace character. However, the new code is only encountered if there was no whitespace or break characters found in step 2.

When is the new code useful?

Reply:
This is the case: Unwanted 1st space on the second line (I'll try to explain it):
![glyphs3](https://user-images.githubusercontent.com/25801199/32605638-6f5d71a2-c552-11e7-8264-4f3ce3d47aad.jpg)

When the line is cut-off and we try to find the index we use the function
```
/** Returns the first valid glyph index to use to wrap to the next line, starting at the specified start index and
 * (typically) moving toward the beginning of the glyphs array. */
public int getWrapIndex (Array<Glyph> glyphs, int start) {
 ...
 return 0;
}
```
In that case when cutting-off the second line the execution falls to the ``return 0;`` last sentence of the function, and the next line starts with a blank space (1st char of the second line). This is because the first char of the ``Array<Glyph> glyphs`` is la blank space.
The new block tries to avoid the ``return 0`` case, advancing the blanks.

You can reproduce the glitch with the sample at issue #4425. Maybe you have to move the "[GREEN]" markup until you catch it.
It occurs when the [GREEN] is before a space. In my sample:
```
String tex2= "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ut lacus[GREEN] maximus, volutpat ex vel, tempor leo....."
```